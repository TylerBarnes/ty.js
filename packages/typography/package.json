{
  "name": "tyjs",
  "description":
    "A fork of Kyle Mathews Typography.js with added breakpoint support",
  "version": "0.0.2",
  "author": "Tyler Barnes <tyler@known.design>",
  "dependencies": {
    "compass-vertical-rhythm": "^1.3.0",
    "decamelize": "^1.2.0",
    "gray-percentage": "^2.0.0",
    "lodash": "^4.13.1",
    "modularscale": "^1.0.2",
    "object-assign": "^4.1.0",
    "typography-normalize": "^0.14.0"
  },
  "keywords": [
    "css",
    "inline styles",
    "react",
    "typography",
    "vertical-rhythm"
  ],
  "license": "MIT",
  "homepage": "https://github.com/TylerBarnes/ty.js",
  "bugs": {
    "url": "https://github.com/TylerBarnes/ty.js/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/TylerBarnes/ty.js.git"
  },
  "main": "dist/index.js",
  "scripts": {
    "build": "../../node_modules/.bin/babel src --out-dir dist/",
    "flow": "../../node_modules/.bin/flow",
    "lint":
      "../../node_modules/.bin/eslint --ext .js,.jsx --ignore-path ../../.gitignore .",
    "watch": "../../node_modules/.bin/babel -w src --out-dir dist/"
  }
}
