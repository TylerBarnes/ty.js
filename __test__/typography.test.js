// @flow

import typography from "../packages/typography/src/index"

describe("typography(options?)", () => {
  it("should be a function", () => {
    expect(typography).toEqual(jasmine.any(Function))
  })

  it("should return a structure", () => {
    const actual = typography()
    expect(actual.options).toEqual(jasmine.any(Object))
    expect(actual.rhythm).toEqual(jasmine.any(Function))
    expect(actual.establishBaseline).toEqual(jasmine.any(Function))
    expect(actual.linesForFontSize).toEqual(jasmine.any(Function))
    expect(actual.adjustFontSizeTo).toEqual(jasmine.any(Function))
    expect(actual.scale).toEqual(jasmine.any(Function))
    expect(actual.createStyles).toEqual(jasmine.any(Function))
    expect(actual.toJSON).toEqual(jasmine.any(Function))
    expect(actual.toString).toEqual(jasmine.any(Function))
    expect(actual.injectStyles).toEqual(jasmine.any(Function))
  })
})

describe("typography(options?).scale()", () => {
  it("should scale", () => {
    const actual = typography().scale()
    expect(actual.fontSize).toEqual("1rem")
    expect(actual.lineHeight).toEqual("1.45rem")
  })

  it("should accept custom scales", () => {
    const actual = typography().scale(1.333)
    expect(actual.fontSize).toEqual("2.51926rem")
    expect(actual.lineHeight).toEqual("2.9rem")
  })

  it("should accept custom scales", () => {
    const actual = typography().scale(2)
    expect(actual.fontSize).toEqual("4rem")
    expect(actual.lineHeight).toEqual("4.35rem")
  })
})

describe("typography(options?).toJSON()", () => {
  it("should return CSS as JSON", () => {
    expect(typography().toJSON()).toMatchSnapshot()
  })
})

describe("typography(options?).toString()", () => {
  it("should return CSS as a string", () => {
    expect(
      typography({
        includeNormalize: false,
      }).toString()
    ).toMatchSnapshot()
  })
})

describe("typography(options?).createStyles()", () => {
  it("should return CSS as a string", () => {
    expect(
      typography({
        includeNormalize: false,
      }).createStyles()
    ).toMatchSnapshot()
  })
})

describe("typography(options?).injectStyles()", () => {
  it("should not fail if document is undefined", () => {
    expect(() => {
      typography().injectStyles()
    }).not.toThrow()
  })

  it("should set css if typography.js element exists", () => {
    const sut = typography()

    global.document = jasmine.createSpyObj("document", ["getElementById"])

    const styleNode = {}

    global.document.getElementById.and.returnValue(styleNode)

    sut.injectStyles()

    expect(styleNode.innerHTML).toEqual(sut.toString())
    expect(global.document.getElementById).toHaveBeenCalledWith("typography.js")

    delete global.document
  })

  it("should create a new style node if typography.js element does not exists", () => {
    const sut = typography()

    global.document = jasmine.createSpyObj("document", [
      "head",
      "createElement",
      "getElementById",
    ])

    const styleNode = {}

    global.document.getElementById.and.returnValue(null)
    global.document.createElement.and.returnValue(styleNode)
    global.document.head.appendChild = jasmine.createSpy("appendChild")

    sut.injectStyles()

    expect(styleNode.id).toEqual("typography.js")
    expect(styleNode.innerHTML).toEqual(sut.toString())
    expect(global.document.createElement).toHaveBeenCalledWith("style")
    expect(global.document.getElementById).toHaveBeenCalledWith("typography.js")
    expect(global.document.head.appendChild).toHaveBeenCalledWith(styleNode)

    delete global.document
  })
})
