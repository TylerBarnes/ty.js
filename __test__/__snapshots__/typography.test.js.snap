// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typography(options?).createStyles() should return CSS as a string 1`] = `"html{font:100%/1.45em 'georgia',serif;box-sizing:border-box;overflow-y:scroll;}*{box-sizing:inherit;}*:before{box-sizing:inherit;}*:after{box-sizing:inherit;}body{color:hsla(0,0%,0%,0.8);font-family:'georgia',serif;font-weight:normal;word-wrap:break-word;font-kerning:normal;-moz-font-feature-settings:\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\";-ms-font-feature-settings:\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\";-webkit-font-feature-settings:\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\";font-feature-settings:\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\";}img{max-width:100%;margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}h1{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;color:inherit;font-family:-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;font-weight:bold;text-rendering:optimizeLegibility;font-size:2rem;line-height:1.1;}h2{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;color:inherit;font-family:-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;font-weight:bold;text-rendering:optimizeLegibility;font-size:1.51572rem;line-height:1.1;}h3{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;color:inherit;font-family:-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;font-weight:bold;text-rendering:optimizeLegibility;font-size:1.31951rem;line-height:1.1;}h4{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;color:inherit;font-family:-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;font-weight:bold;text-rendering:optimizeLegibility;font-size:1rem;line-height:1.1;}h5{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;color:inherit;font-family:-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;font-weight:bold;text-rendering:optimizeLegibility;font-size:0.87055rem;line-height:1.1;}h6{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;color:inherit;font-family:-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;font-weight:bold;text-rendering:optimizeLegibility;font-size:0.81225rem;line-height:1.1;}hgroup{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}ul{margin-left:1.45rem;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;list-style-position:outside;list-style-image:none;}ol{margin-left:1.45rem;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;list-style-position:outside;list-style-image:none;}dl{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}dd{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}p{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}figure{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}pre{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;font-size:0.85rem;line-height:1.45rem;}table{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;font-size:1rem;line-height:1.45rem;border-collapse:collapse;width:100%;}fieldset{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}blockquote{margin-left:1.45rem;margin-right:1.45rem;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}form{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}noscript{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}iframe{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}hr{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:calc(1.45rem - 1px);background:hsla(0,0%,0%,0.2);border:none;height:1px;}address{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}b{font-weight:bold;}strong{font-weight:bold;}dt{font-weight:bold;}th{font-weight:bold;}li{margin-bottom:calc(1.45rem / 2);}ol li{padding-left:0;}ul li{padding-left:0;}li > ol{margin-left:1.45rem;margin-bottom:calc(1.45rem / 2);margin-top:calc(1.45rem / 2);}li > ul{margin-left:1.45rem;margin-bottom:calc(1.45rem / 2);margin-top:calc(1.45rem / 2);}blockquote *:last-child{margin-bottom:0;}li *:last-child{margin-bottom:0;}p *:last-child{margin-bottom:0;}li > p{margin-bottom:calc(1.45rem / 2);}code{font-size:0.85rem;line-height:1.45rem;}kbd{font-size:0.85rem;line-height:1.45rem;}samp{font-size:0.85rem;line-height:1.45rem;}abbr{border-bottom:1px dotted hsla(0,0%,0%,0.5);cursor:help;}acronym{border-bottom:1px dotted hsla(0,0%,0%,0.5);cursor:help;}abbr[title]{border-bottom:1px dotted hsla(0,0%,0%,0.5);cursor:help;text-decoration:none;}thead{text-align:left;}td,th{text-align:left;border-bottom:1px solid hsla(0,0%,0%,0.12);font-feature-settings:\\"tnum\\";-moz-font-feature-settings:\\"tnum\\";-ms-font-feature-settings:\\"tnum\\";-webkit-font-feature-settings:\\"tnum\\";padding-left:0.96667rem;padding-right:0.96667rem;padding-top:0.725rem;padding-bottom:calc(0.725rem - 1px);}th:first-child,td:first-child{padding-left:0;}th:last-child,td:last-child{padding-right:0;}"`;

exports[`typography(options?).toJSON() should return CSS as JSON 1`] = `
Object {
  "*": Object {
    "boxSizing": "inherit",
  },
  "*:after": Object {
    "boxSizing": "inherit",
  },
  "*:before": Object {
    "boxSizing": "inherit",
  },
  "abbr": Object {
    "borderBottom": "1px dotted hsla(0,0%,0%,0.5)",
    "cursor": "help",
  },
  "abbr[title]": Object {
    "borderBottom": "1px dotted hsla(0,0%,0%,0.5)",
    "cursor": "help",
    "textDecoration": "none",
  },
  "acronym": Object {
    "borderBottom": "1px dotted hsla(0,0%,0%,0.5)",
    "cursor": "help",
  },
  "address": Object {
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "b": Object {
    "fontWeight": "bold",
  },
  "blockquote": Object {
    "marginBottom": "1.45rem",
    "marginLeft": "1.45rem",
    "marginRight": "1.45rem",
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "blockquote *:last-child": Object {
    "marginBottom": 0,
  },
  "body": Object {
    "MozFontFeatureSettings": "\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\"",
    "WebkitFontFeatureSettings": "\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\"",
    "color": "hsla(0,0%,0%,0.8)",
    "fontFamily": "'georgia',serif",
    "fontFeatureSettings": "\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\"",
    "fontKerning": "normal",
    "fontWeight": "normal",
    "msFontFeatureSettings": "\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\"",
    "wordWrap": "break-word",
  },
  "code": Object {
    "fontSize": "0.85rem",
    "lineHeight": "1.45rem",
  },
  "dd": Object {
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "dl": Object {
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "dt": Object {
    "fontWeight": "bold",
  },
  "fieldset": Object {
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "figure": Object {
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "form": Object {
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "h1": Object {
    "color": "inherit",
    "fontFamily": "-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif",
    "fontSize": "2rem",
    "fontWeight": "bold",
    "lineHeight": 1.1,
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "textRendering": "optimizeLegibility",
  },
  "h2": Object {
    "color": "inherit",
    "fontFamily": "-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif",
    "fontSize": "1.51572rem",
    "fontWeight": "bold",
    "lineHeight": 1.1,
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "textRendering": "optimizeLegibility",
  },
  "h3": Object {
    "color": "inherit",
    "fontFamily": "-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif",
    "fontSize": "1.31951rem",
    "fontWeight": "bold",
    "lineHeight": 1.1,
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "textRendering": "optimizeLegibility",
  },
  "h4": Object {
    "color": "inherit",
    "fontFamily": "-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif",
    "fontSize": "1rem",
    "fontWeight": "bold",
    "lineHeight": 1.1,
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "textRendering": "optimizeLegibility",
  },
  "h5": Object {
    "color": "inherit",
    "fontFamily": "-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif",
    "fontSize": "0.87055rem",
    "fontWeight": "bold",
    "lineHeight": 1.1,
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "textRendering": "optimizeLegibility",
  },
  "h6": Object {
    "color": "inherit",
    "fontFamily": "-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif",
    "fontSize": "0.81225rem",
    "fontWeight": "bold",
    "lineHeight": 1.1,
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "textRendering": "optimizeLegibility",
  },
  "hgroup": Object {
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "hr": Object {
    "background": "hsla(0,0%,0%,0.2)",
    "border": "none",
    "height": "1px",
    "marginBottom": "calc(1.45rem - 1px)",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "html": Object {
    "boxSizing": "border-box",
    "font": "100%/1.45em 'georgia',serif",
    "overflowY": "scroll",
  },
  "iframe": Object {
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "img": Object {
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "maxWidth": "100%",
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "kbd": Object {
    "fontSize": "0.85rem",
    "lineHeight": "1.45rem",
  },
  "li": Object {
    "marginBottom": "calc(1.45rem / 2)",
  },
  "li *:last-child": Object {
    "marginBottom": 0,
  },
  "li > ol": Object {
    "marginBottom": "calc(1.45rem / 2)",
    "marginLeft": "1.45rem",
    "marginTop": "calc(1.45rem / 2)",
  },
  "li > p": Object {
    "marginBottom": "calc(1.45rem / 2)",
  },
  "li > ul": Object {
    "marginBottom": "calc(1.45rem / 2)",
    "marginLeft": "1.45rem",
    "marginTop": "calc(1.45rem / 2)",
  },
  "noscript": Object {
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "ol": Object {
    "listStyleImage": "none",
    "listStylePosition": "outside",
    "marginBottom": "1.45rem",
    "marginLeft": "1.45rem",
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "ol li": Object {
    "paddingLeft": 0,
  },
  "p": Object {
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "p *:last-child": Object {
    "marginBottom": 0,
  },
  "pre": Object {
    "fontSize": "0.85rem",
    "lineHeight": "1.45rem",
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "samp": Object {
    "fontSize": "0.85rem",
    "lineHeight": "1.45rem",
  },
  "strong": Object {
    "fontWeight": "bold",
  },
  "table": Object {
    "borderCollapse": "collapse",
    "fontSize": "1rem",
    "lineHeight": "1.45rem",
    "marginBottom": "1.45rem",
    "marginLeft": 0,
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
    "width": "100%",
  },
  "td,th": Object {
    "MozFontFeatureSettings": "\\"tnum\\"",
    "WebkitFontFeatureSettings": "\\"tnum\\"",
    "borderBottom": "1px solid hsla(0,0%,0%,0.12)",
    "fontFeatureSettings": "\\"tnum\\"",
    "msFontFeatureSettings": "\\"tnum\\"",
    "paddingBottom": "calc(0.725rem - 1px)",
    "paddingLeft": "0.96667rem",
    "paddingRight": "0.96667rem",
    "paddingTop": "0.725rem",
    "textAlign": "left",
  },
  "th": Object {
    "fontWeight": "bold",
  },
  "th:first-child,td:first-child": Object {
    "paddingLeft": 0,
  },
  "th:last-child,td:last-child": Object {
    "paddingRight": 0,
  },
  "thead": Object {
    "textAlign": "left",
  },
  "ul": Object {
    "listStyleImage": "none",
    "listStylePosition": "outside",
    "marginBottom": "1.45rem",
    "marginLeft": "1.45rem",
    "marginRight": 0,
    "marginTop": 0,
    "paddingBottom": 0,
    "paddingLeft": 0,
    "paddingRight": 0,
    "paddingTop": 0,
  },
  "ul li": Object {
    "paddingLeft": 0,
  },
}
`;

exports[`typography(options?).toString() should return CSS as a string 1`] = `"html{font:100%/1.45em 'georgia',serif;box-sizing:border-box;overflow-y:scroll;}*{box-sizing:inherit;}*:before{box-sizing:inherit;}*:after{box-sizing:inherit;}body{color:hsla(0,0%,0%,0.8);font-family:'georgia',serif;font-weight:normal;word-wrap:break-word;font-kerning:normal;-moz-font-feature-settings:\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\";-ms-font-feature-settings:\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\";-webkit-font-feature-settings:\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\";font-feature-settings:\\"kern\\", \\"liga\\", \\"clig\\", \\"calt\\";}img{max-width:100%;margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}h1{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;color:inherit;font-family:-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;font-weight:bold;text-rendering:optimizeLegibility;font-size:2rem;line-height:1.1;}h2{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;color:inherit;font-family:-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;font-weight:bold;text-rendering:optimizeLegibility;font-size:1.51572rem;line-height:1.1;}h3{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;color:inherit;font-family:-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;font-weight:bold;text-rendering:optimizeLegibility;font-size:1.31951rem;line-height:1.1;}h4{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;color:inherit;font-family:-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;font-weight:bold;text-rendering:optimizeLegibility;font-size:1rem;line-height:1.1;}h5{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;color:inherit;font-family:-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;font-weight:bold;text-rendering:optimizeLegibility;font-size:0.87055rem;line-height:1.1;}h6{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;color:inherit;font-family:-apple-system,'BlinkMacSystemFont','Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;font-weight:bold;text-rendering:optimizeLegibility;font-size:0.81225rem;line-height:1.1;}hgroup{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}ul{margin-left:1.45rem;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;list-style-position:outside;list-style-image:none;}ol{margin-left:1.45rem;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;list-style-position:outside;list-style-image:none;}dl{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}dd{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}p{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}figure{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}pre{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;font-size:0.85rem;line-height:1.45rem;}table{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;font-size:1rem;line-height:1.45rem;border-collapse:collapse;width:100%;}fieldset{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}blockquote{margin-left:1.45rem;margin-right:1.45rem;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}form{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}noscript{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}iframe{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}hr{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:calc(1.45rem - 1px);background:hsla(0,0%,0%,0.2);border:none;height:1px;}address{margin-left:0;margin-right:0;margin-top:0;padding-bottom:0;padding-left:0;padding-right:0;padding-top:0;margin-bottom:1.45rem;}b{font-weight:bold;}strong{font-weight:bold;}dt{font-weight:bold;}th{font-weight:bold;}li{margin-bottom:calc(1.45rem / 2);}ol li{padding-left:0;}ul li{padding-left:0;}li > ol{margin-left:1.45rem;margin-bottom:calc(1.45rem / 2);margin-top:calc(1.45rem / 2);}li > ul{margin-left:1.45rem;margin-bottom:calc(1.45rem / 2);margin-top:calc(1.45rem / 2);}blockquote *:last-child{margin-bottom:0;}li *:last-child{margin-bottom:0;}p *:last-child{margin-bottom:0;}li > p{margin-bottom:calc(1.45rem / 2);}code{font-size:0.85rem;line-height:1.45rem;}kbd{font-size:0.85rem;line-height:1.45rem;}samp{font-size:0.85rem;line-height:1.45rem;}abbr{border-bottom:1px dotted hsla(0,0%,0%,0.5);cursor:help;}acronym{border-bottom:1px dotted hsla(0,0%,0%,0.5);cursor:help;}abbr[title]{border-bottom:1px dotted hsla(0,0%,0%,0.5);cursor:help;text-decoration:none;}thead{text-align:left;}td,th{text-align:left;border-bottom:1px solid hsla(0,0%,0%,0.12);font-feature-settings:\\"tnum\\";-moz-font-feature-settings:\\"tnum\\";-ms-font-feature-settings:\\"tnum\\";-webkit-font-feature-settings:\\"tnum\\";padding-left:0.96667rem;padding-right:0.96667rem;padding-top:0.725rem;padding-bottom:calc(0.725rem - 1px);}th:first-child,td:first-child{padding-left:0;}th:last-child,td:last-child{padding-right:0;}"`;
